src/main/java/org/cloudfoundry/promregator/config/PromregatorConfiguration.kt

```diff
@@ -0,0 +1,88 @@
+package org.cloudfoundry.promregator.config
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+import java.util.*
+
+@ConstructorBinding
+@ConfigurationProperties(prefix = "promregator", ignoreInvalidFields = true)
+data class PromregatorConfiguration(
+
+        val workaround: WorkAroundConfig = WorkAroundConfig(),
+        val simulation: SimulationConfig = SimulationConfig(),
+        val reactor: ReactorConfig = ReactorConfig(),
+
+        var targets: List<Target> = ArrayList(),
+        var authenticator: AuthenticatorConfiguration = AuthenticatorConfiguration(),
+        var targetAuthenticators: List<AuthenticatorConfiguration> = ArrayList(),
+        val scraping: ScrapingConfig = ScrapingConfig(),
+        val metrics: MetricsConfig = MetricsConfig(),
+        val discoverer: DiscovererConfig = DiscovererConfig(),
+        val discovery: DiscoveryConfig = DiscoveryConfig(),
+        val endpoint: EndpointConfig = EndpointConfig(),
+        val cache: PromCacheConfig = PromCacheConfig()
+)
+
+open class AuthenticatorConfiguration(
```

Comment:
Just out of curiosity: Why do you have to declare this as `open class` and not as `data class`?

Reply:
Oops! That was because the `TargetAutheticatorConfiguration` used to extend from this one. I'll switch it back to a `data class`.  I decided to unify them and just give the root one a default ID